### PR TITLE
Composite checkout: fix style leaks on stripe fields

### PR DESCRIPTION
**Pull request by fditrapani**
#### Changes proposed in this Pull Request

This PR updates the styles of our composite checkout fields to prevent leakage from calypso.

**Before**
Screeny Video Dec 2, 2019 at 10 59 55 AM

**After**
Screeny Video Dec 2, 2019 at 11 03 59 AM

#### Testing instructions

Visit checkout with the store sandboxed. Add the `composite-checkout-wpcom` flag. Keep an eye on the fields as they load. You shouldn't see any flickering.

### CODE REVIEW

**Comment by fditrapani** [comment]
Sounds good. Let me know what you find. I tried adding more specificity in the definition by adding `[type="text]` (which reads as `.sdkjsd[type='text']` ) but Calypso is defining styles right on the elements which makes it difficult to override.

**Comment by sirbrillig** [comment]
I was actually going to say... Calypso is supposed to be using BEM too, as far as I know. Maybe there's something we can fix in Calypso instead of here?

**Comment by fditrapani** [comment]
> Maybe there's something we can fix in Calypso instead of here?

That's not a bad idea. I know where to make that change but it's kind of a scary change because of how broadly this is being applied right now. I don't know if there would be a reliable way of testing it across the board.

**Comment by fditrapani** [comment]
@sirbrillig I found another way. I was missing a `&` when I tried to overide the styles with emotion. This latest update works without needing to touch Calypso or use a `!important` statement.

File: packages/composite-checkout/src/components/field.js [hunk 1]
```diff
@@ -109,9 +109,15 @@ const Input = styled.input`
 	width: 100%;
 	box-sizing: border-box;
 	font-size: 16px;
-	border: 1px solid
-		${props => ( props.isError ? props.theme.colors.error : props.theme.colors.borderColor )};
-	padding: 13px ${props => ( props.icon ? '60px' : '10px' )} 12px 10px;
+
+	//override leaky styles with important
```
Comment (sirbrillig) [inline packages/composite-checkout/src/components/field.js]: this comment isn't quite accurate any more.

**Pull request merged**